### PR TITLE
Grid

### PR DESCRIPTION
**Pull request by bvdhoek**
CircuitModel.remove and CircuitModel.isCompleteCircle need to be changed/implemented, but was unnecessary for the grid

### CODE REVIEW

**Comment by bvdhoek** [comment]
CircuitModel.remove and CircuitModel.isCompleteCircle need to be changed/implemented, but was unneccesary for the grid

File: app/src/main/java/com/edulectronics/tinycircuit/Circuit/CircuitController.java [hunk 1]
```diff
@@ -2,41 +2,41 @@
...
+            circuit.add(component, x, y);
         }
     }
 
     public void removeComponent(Component c, int x, int y){
-        if(occupation[x][y]) {
-            occupation[x][y] = false;
-            circuitModel.components.remove(c);
+        if(circuit.occupied(x, y)) {
+            circuit.remove(c);
```
Comment (MaaikeG) [inline app/src/main/java/com/edulectronics/tinycircuit/Circuit/CircuitController.java]: This is dangerous. Remove an item only based on coordinates, or only based on object. Not both

File: app/src/main/java/com/edulectronics/tinycircuit/Circuit/CircuitModel.java [hunk 2]
```diff
@@ -1,32 +1,59 @@
...
 
     /*Checks if the Circuit has a connection from the first powersource*/
-    public boolean checkConnection(){
-        for (Component c : components){
-            if (c.getClass() == Powersource.class){
-                if (c.hasOutputConnection()){
-                    return true;
-                }
+    public boolean noComponentsWithoutOutput(){
+        for (Object object: components) {
```
Comment (MaaikeG) [inline app/src/main/java/com/edulectronics/tinycircuit/Circuit/CircuitModel.java]: Object should be component.

File: app/src/main/java/com/edulectronics/tinycircuit/Circuit/CircuitModel.java [hunk 3]
```diff
@@ -1,32 +1,59 @@
...
+            if (object != null) {
+                Component c = (Component) object;
+                if (!c.hasOutputConnection()) { return false; }
```
Comment (MaaikeG) [inline app/src/main/java/com/edulectronics/tinycircuit/Circuit/CircuitModel.java]: This only needs to be done for the power source! hasOutputConnection() iterates over all components automatically

File: app/src/main/java/com/edulectronics/tinycircuit/Circuit/CircuitModel.java [hunk 4]
```diff
@@ -1,32 +1,59 @@
...
+
+    public boolean isCompleteCircle() {
         return false;
     }
+
+    public void add(Component component, int x, int y) {
+        components[x][y] = component;
+    }
+
+    public void remove(Component component) {
```
Comment (MaaikeG) [inline app/src/main/java/com/edulectronics/tinycircuit/Circuit/CircuitModel.java]: should remove based on x and y value

**Pull request merged**